AdminCore.FsmWorkflow/FsmMachines/WorkflowFsmPto.cs

```diff
@@ -0,0 +1,150 @@
+using System.Collections.Generic;
+using AdminCore.Common;
+using AdminCore.Constants.Enums;
+using AdminCore.FsmWorkflow.FsmMachines.FsmLeaveStates;
+using AdminCore.FsmWorkflow.FsmMachines.FsmLeaveTriggers;
+using AdminCore.FsmWorkflow.FsmMachines.FsmWorkflowState;
+using Stateless;
+
+namespace AdminCore.FsmWorkflow.FsmMachines
+{
+    public class WorkflowFsmPto : WorkflowFsm<PtoState, LeaveTriggersPto>
+    {       
+        private StateMachine<PtoState, LeaveTriggersPto>.TriggerWithParameters<EventStatuses, string> LeaveResponseTrigger;
+        
+        public WorkflowFsmPto(WorkflowStateData fsmStateData)
+        {
+            ConfigureFsm(fsmStateData);
+        }
+        
+        public override void ConfigureFsm(WorkflowStateData fsmStateData)
+        {
+            FsmStateData = fsmStateData;
+            FsMachine = new StateMachine<PtoState, LeaveTriggersPto>(() => (PtoState)FsmStateData.CurrentState, s => FsmStateData.CurrentState = (int)s);
+            
+            LeaveResponseTrigger = FsMachine.SetTriggerParameters<EventStatuses, string>(LeaveTriggersPto.LeaveResponded);
+            
+            // Leave Awaiting Team Lead and Client
+            FsMachine.Configure(PtoState.LeaveAwaitingTeamLeadClient)
+                .SubstateOf(PtoState.LeaveAwaitingResponses)
+                .OnActivate(() =>
+                {
+                    if (IsAdminResponseReceived(FsmStateData.ApprovalDict))
+                    {
+                        FsMachine.Fire(IsAdminResponseApprove(FsmStateData.ApprovalDict)
```

Comment:
same situation, should we be reusing the result of zIsAdminResponseApprove`

Reply:
OnActivate is a callback similar to OnEntry. "Activation events will fire when transitioning into the state, and when the Activate() method is called."

I called Activate() when instantiating a workflow to enforce the OnActivate transition check to ensure the correct state.

```
bool isAdminRepRec;
State1.OnActivate(() => isAdminRepRec = IsAdminResponseReceived(FsmStateData.ApprovalDict))
State2.OnActivate(() => isAdminRepRec
```

It's possible a trigger can fire where the FSMs initial state is State2 and never pass through State1 to setup the result of IsAdminResponseReceived.
I think it is safer to keep it the way it is.